pkg/shortnames/shortnames.go

```diff
@@ -386,6 +380,39 @@ func Resolve(ctx *types.SystemContext, name string) (*Resolved, error) {
 	return resolved, nil
 }
 
+// Prompt opens a prompt in the terminal to let the user choose *one* of the
+// specified items.
+func Prompt(items []string) (string, error) {
+	switch len(items) {
+	case 0:
+		return fmt.Errorf("cannot prompt without items")
+	case 1:
+		return items[0]
+	default:
+		if !term.IsTerminal(int(os.Stdout.Fd())) || !term.IsTerminal(int(os.Stdin.Fd())) {
```

Comment:
If we take the effort to avoid repeating the two syscalls for this check for the in-package caller, it might be worth it factoring the TTY check into a small helper function, to ensure both callers of `promptNoTTYCheck` do _exactly_ the same check and they don’t drift apart over time.